fs/io_zenfs.cc

```diff
@@ -611,6 +623,26 @@ IOStatus ZoneFile::Recover() {
   return IOStatus::OK();
 }
 
+void ZoneFile::ReplaceExtentList(std::vector<ZoneExtent*> new_list) {
+  assert(!IsOpenForWR() && new_list.size() > 0);
+  assert(new_list.size() == extents_.size());
+
+  WriteLock lck(this);
+
+  // Clear changed extents' zone stats
+  for (int i = 0; i < new_list.size(); ++i) {
+    ZoneExtent* old_ext = extents_[i];
+    if (old_ext->start_ != new_list[i]->start_) {
+      old_ext->zone_->used_capacity_ -= old_ext->length_;
+    }
+    delete old_ext;
+  }
+
+  extents_ = new_list;
+  ZoneExtent* last = new_list.back();
+  extent_start_ = last->start_;
```

Comment:
Why do you set extent_start_ here? That should be NO_EXTENT for closed files..

Reply:
Fixed, thanks.